File: Python/7 kyu Get the Middle Character/get_middle.py
# You are going to be given a word. Your job is to return the middle character
# of the word. If the word's length is odd, return the middle character.
# If the word's length is even, return the middle 2 characters.

# Examples:
# get_middle("test") should return "es"
# get_middle("testing") should return "t"
# get_middle("middle") should return "dd"
# get_middle("A") should return "A"

# My solution

def get_middle(s):
    mid = len(s) // 2
    return s[mid] if len(s) % 2 else s[mid - 1:mid + 1]

File: Php/7 kyu Get the Middle Character/getMiddle.php
/*
You are going to be given a word. Your job is to return the middle character
of the word. If the word's length is odd, return the middle character.
If the word's length is even, return the middle 2 characters.

Examples:
getMiddle("test") should return "es"
getMiddle("testing") should return "t"
getMiddle("middle") should return "dd"
getMiddle("A") should return "A"
*/

// Solution

function getMiddle(string $s): string {
    $mid = intdiv(strlen($s), 2);
    if (strlen($s) % 2 === 0) {
        return substr($s, $mid - 1, 2);
    }
    return substr($s, $mid, 1);
}
